fs/_ftp_parse.py

```diff
@@ -82,12 +85,12 @@ def parse_line(line):
 
 
 def _parse_time(t, formats):
-    t = " ".join(token.strip() for token in t.lower().split(" "))
-
     _t = None
     for frmt in formats:
         try:
             _t = time.strptime(t, frmt)
+            if t:
```

Comment:
I suspect you meant to do `if _t:` here?

Reply:
True. But I guess this just shows that the line is entirely noop since it wasn't ever reached during the tests. Are you sure you want to have it in there? Then I'll commit a fix for the typo.

Reply:
As I mentioned in an earlier comment on this issue, the current code (and tests) assume that in the list of `formats`, there'll only be a single format that doesn't raise a `ValueError`. But IMHO it makes sense to (also) stop looping after the first "valid" format, rather than continuing to loop through all of them?

Reply:
Sure, I understand that. My suggestion would be to just omit the `if _t` as, when that piece of code is reached, `_t` will always have a valid value.